arm-servicebus/2014-09-01/swagger/servicebus.json

```diff
@@ -566,153 +572,2046 @@
+        },
+        "EnableDeadLetteringOnMessageExpiration": {
+          "type": "boolean",
+          "description": "Value that indicates if a subscription has dead letter support when a message expires."
+        },
+        "ExtensionData": {
+          "type": "object",
+          "description": "Structure that contains extra data."
+        },
+        "ForwardDeadLetteredMessagesTo": {
+          "type": "string",
+          "description": "Path to the recipient to which the dead lettered messages are forwarded."
+        },
+        "ForwardTo": {
+          "type": "string",
+          "description": "Path to the recipient to which the dead lettered messages are forwarded."
+        },
+        "IsReadOnly": {
+          "type": "boolean",
+          "description": "Value that indicates whether the entity description is read-only."
+        },
+        "LockDuration": {
+          "type": "string",
+          "format": "time-span",
+          "description": "The lock duration time span for the subscription."
+        },
+        "MaxDeliveryCount": {
+          "format": "int32",
+          "type": "integer",
+          "description": "Number of maximum deliveries."
+        },
+        "MessageCount": {
+          "format": "int64",
+          "type": "integer",
+          "description": "Number of messages."
+        },
+        "MessageCountDetails": {
+          "type": "object",
+          "description": "Message details for a subscription."
+        },
+        "Name": {
+          "type": "string",
+          "description": "Name of the subscription description."
+        },
+        "RequiresSession": {
+          "type": "boolean",
+          "description": "Value indicating if a subscription supports the concept of session."
+        },
+        "EntityStatus": {
+          "type": "string",
+          "description": "Enumerates the possible values for the status of a messaging entity.",
+          "enum": [
+            "Active",
+            "Creating",
+            "Deleting",
+            "Disabled",
+            "ReceiveDisabled",
+            "Renaming",
+            "Restoring",
+            "SendDisabled",
+            "Unknown"
+          ],
+          "x-ms-enum": {
+            "name": "EntityStatus",
+            "modelAsString": false
+          }
+        },
+        "TopicPath": {
+          "type": "string",
+          "description": "Path of the topic that this subscription description belongs to."
+        },
+        "UpdatedAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The exact time the message has been updated."
+        },
+        "UserMetadata": {
+          "type": "string",
+          "description": "The user metadata."
+        }
+      },
+	  "x-ms-client-flatten": true,
+      "description": "Represents the metadata description of the Subscription."
+    },
+    "SubscriptionResource": {
```

Comment:
Why can you make this an allof on the "Resource" definition, as "Resource" is already defined with almost same top level properties.

Please take a look at [the definition of "StorageAccount" in the storage management spec](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-01-01/swagger/storage.json#L769) to get an idea.